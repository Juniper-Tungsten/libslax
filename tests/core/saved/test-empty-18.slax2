version 1.2;

ns my = "http://my.example.com/my";

function my:test ($x) {
    result {
        <test> $x;
    }
}

match foo[../../id/text() _ "/" _ text()] {
    var $test2 = "€";
    var $test = "";
    /* This one is odd */
    expr "text in " _ name(..) _ ' = "';
    /* This one too */
    var $date = substring(date, 1, 4) _ "/" _ substring(date, 5, 2) _ "/" _ substring(date, 7, 2);
    /* This one tickles the bug */
    expr my:test(local-game-prefix _ opponent, answer);
    /* This one tickles a second bug */
    <span class="element"> "</" _ name(.) _ ">";
}

main <top> {
    var $test := my:test("fish");
    var $this := <that> "one";
    var $that := <your> "was";
    var $gray := <new> "black";
    
    copy-of $this;
    copy-of $that;
    copy-of $gray;
}
