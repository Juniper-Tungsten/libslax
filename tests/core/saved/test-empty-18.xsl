<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:slax-func="http://exslt.org/functions" xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" version="1.0" extension-element-prefixes="slax-func slax-ext">
  <slax-func:function xmlns:slax-func="http://exslt.org/functions" name="test">
    <xsl:param name="x"/>
    <slax-func:result xmlns:slax-func="http://exslt.org/functions">
      <test>
        <xsl:value-of select="$x"/>
      </test>
    </slax-func:result>
  </slax-func:function>
  <xsl:template match="foo[concat(../../id/text(), &quot;/&quot;, text())]">
    <xsl:variable name="test2" select="&quot;€&quot;"/>
    <xsl:variable name="test" select="&quot;&quot;"/>
    <!-- This one is odd -->
    <xsl:value-of select="concat(&quot;text in &quot;, name(..), ' = &quot;')"/>
    <!-- This one too -->
    <xsl:variable name="date" select="concat(substring(date, 1, 4), &quot;/&quot;, substring(date, 5, 2), &quot;/&quot;, substring(date, 7, 2))"/>
    <!-- This one tickles the bug -->
    <xsl:value-of select="test(concat(local-game-prefix, opponent), answer)"/>
    <!-- This one tickles a second bug -->
    <span class="element">
      <xsl:value-of select="concat(&quot;&lt;/&quot;, name(.), &quot;&gt;&quot;)"/>
    </span>
  </xsl:template>
  <xsl:template match="/">
    <top>
      <xsl:variable xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" name="test" select="slax-ext:node-set(test(&quot;fish&quot;))"/>
      <xsl:variable name="this-temp-1">
        <that>one</that>
      </xsl:variable>
      <xsl:variable xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" name="this" select="slax-ext:node-set($this-temp-1)"/>
      <xsl:variable name="mine-temp-2">
        <your>was</your>
      </xsl:variable>
      <xsl:variable xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" name="mine" select="slax-ext:node-set($mine-temp-2)"/>
      <xsl:variable name="gray-temp-3">
        <new>black</new>
      </xsl:variable>
      <xsl:variable xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" name="gray" select="slax-ext:node-set($gray-temp-3)"/>
      <xsl:copy-of select="$this"/>
      <xsl:copy-of select="$that"/>
      <xsl:copy-of select="$gray"/>
    </top>
  </xsl:template>
</xsl:stylesheet>
