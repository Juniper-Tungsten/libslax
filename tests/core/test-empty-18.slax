version 1.1;

ns my = "http://my.example.com/my";

function my:test ($x) {
    result <test> $x;
}

match foo[concat(../../id/text(), "/", text())] {

    var $test2 = "\u+20ac";
    var $test = "\u+e004";

    /* This one is odd */
    expr "text in " _ name(..) _ ' = "';

    /* This one too */
    var $date = substring(date, 1, 4)  _ "/"
	_ substring(date, 5, 2)  _ "/" _ substring(date, 7, 2);

    /* This one tickles the bug */
    expr my:test( concat( local-game-prefix, opponent ), answer );

    /* This one tickles a second bug */
    <span class="element"> "</" _ name(.) _ ">";
}

match / {
    <top> {
	var $test := my:test("fish");
	var $this := <that> "one";
	var $that := <your> "was";
	var $gray := <new> "black";
	copy-of $this;
	copy-of $that;
	copy-of $gray;
    }
}