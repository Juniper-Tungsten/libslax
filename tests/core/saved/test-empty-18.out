<?xml version="1.0"?>
<top xmlns:my="http://my.example.com/my">
  <that>one</that>
  <your>was</your>
  <new>black</new>
</top>
